dipy/tracking/local/optimizedlocaltracking.py

```diff
@@ -0,0 +1,420 @@
+        if affine.shape != (4, 4):
+            raise ValueError("affine should be a (4, 4) array.")
+        self.affine = affine
+        self._voxel_size = self._get_voxel_size(affine)
+        self.step_size = step_size
+        self.fixed = fixedstep
+        self.max_cross = max_cross
+        self.maxlen = maxlen
+        self.return_all = return_all
+
+    
+    def pickle_test():
+        pass
+    
+    def __iter__(self):
+        # Make tracks, move them to point space and return
+        track = self._generate_streamlines()
+        return utils.move_streamlines(track, self.affine)
+
+    def _generate_streamlines(self):
+        """A streamline generator"""
+        N = self.maxlen
+        dg = self.direction_getter
+        tc = self.tissue_classifier
+        ss = self.step_size
+        fixed = self.fixed
+        max_cross = self.max_cross
+        vs = self._voxel_size
+
+        # Get inverse transform (lin/offset) for seeds
+        inv_A = np.linalg.inv(self.affine)
+        lin = inv_A[:3, :3]
+        offset = inv_A[:3, 3]
+
+        F = np.empty((N + 1, 3), dtype=float)
+        B = F.copy()
+        for s in self.seeds:
+            s = np.dot(lin, s) + offset
+            directions = dg.initial_direction(s)
+            directions = directions[:max_cross]
+            for first_step in directions:
+                stepsF, tissue_class = local_tracker(dg, tc, s, first_step,
+                                                     vs, F, ss, fixed)
+                if not (self.return_all or
+                        tissue_class == TissueTypes.ENDPOINT or
+                        tissue_class == TissueTypes.OUTSIDEIMAGE):
+                    continue
+                first_step = -first_step
+                stepsB, tissue_class = local_tracker(dg, tc, s, first_step,
+                                                     vs, B, ss, fixed)
+                if not (self.return_all or
+                        tissue_class == TissueTypes.ENDPOINT or
+                        tissue_class == TissueTypes.OUTSIDEIMAGE):
+                    continue
+
+                if stepsB == 1:
+                    streamline = F[:stepsF].copy()
+                else:
+                    parts = (B[stepsB-1:0:-1], F[:stepsF])
+                    streamline = np.concatenate(parts, axis=0)
+                yield streamline
+ 
+    @staticmethod
+    def bunch_arguments(inv_A, lin, offset, F, B, vs, dg, tc, affine,
+                 ss, max_cross=None, maxlen=500, fixed=True,
+                 return_all=True):
+        args = []
+        args.append(inv_A)
+        args.append(lin)
+        args.append(offset)
+        args.append(F)
+        args.append(B)
+        args.append(vs)
+        args.append(dg)
+        args.append(tc)
+        args.append(affine)
+        args.append(ss)
+        args.append(max_cross)
+        args.append(maxlen)
+        args.append(fixed)
+        args.append(return_all)
+        return args
```

Comment:
this is easier to read if you right `args = [inv_A, lin, offset, ..., return_all]`.